qiskit/circuit/library/n_local/qaoa_ansatz.py

```diff
@@ -234,3 +242,33 @@ def mixer_operator(self, mixer_operator) -> None:
         """
         self._mixer = mixer_operator
         self._invalidate()
+
+    def _build(self):
+        if self._data is not None:
+            return
+
+        super()._build()
+
+        # keep old parameter order: first cost operator, then mixer operators
+        num_cost = 0 if _is_pauli_identity(self.cost_operator) else 1
+        if isinstance(self.mixer_operator, QuantumCircuit):
+            num_mixer = self.mixer_operator.num_parameters
+        else:
+            num_mixer = 0 if _is_pauli_identity(self.mixer_operator) else 1
+
+        num_parameters = self.reps * (num_cost + num_mixer)
+        theta = ParameterVector("θ", num_parameters)
+
+        # 0 3 1 4 2 5
+        # 0 4 5 1 6 7 3
+        # first the cost operator parameters then mixer operators
+        reordered = [None] * num_parameters
+        theta_iter = iter(theta)
+
+        reordered = [next(theta_iter) for _ in range(self.reps * num_cost)]
+
+        for i in range(self.reps):
+            start_idx = i * num_mixer + (i + 1) * num_cost
+            reordered[start_idx:start_idx] = [next(theta_iter) for _ in range(num_mixer)]
```

Comment:
This is quite an unusual way of making a permutation - I wasn't able to figure it out just by reading it.  Would something like
```python
        # Create a permutation to take us from (cost_1, mixer_1, cost_2, mixer_2, ...)
        # to (cost_1, cost_2, ..., mixer_1, mixer_2, ...).
        total_cost = self.reps * num_cost
        reordered = []
        for rep in range(self.reps):
            reordered.extend(theta[rep * num_cost : (rep + 1) * num_cost])
            reordered.extend(
                theta[total_cost + rep * num_mixer : total_cost + (rep + 1) * num_mixer]
            )
```
be clearer?  (assuming it's the same as intended!)

Reply:
It's the same yes, I'll replace it for your snippet 👍🏻